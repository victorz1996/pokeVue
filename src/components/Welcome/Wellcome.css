
.img {
    max-height: 300px;
}
.boton {
    background: #f22539;
    border-radius: 10rem;
    color: aliceblue;
    height: 50px;
    width: 120px;
}