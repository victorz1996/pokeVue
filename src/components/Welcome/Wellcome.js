/**
 * @namespace Wellcome
 * @description Componente que contiene la informacion que se muestra en el inicio de la
 * aplicacion
 * @author       Victor Zabala <victorma164@gmail.com>
 */
export default {
    name: 'Wellcome',
    methods: {
      /**
        * @memberOf Wellcome
        * @function iniciar inicio de la app
        * @description Metodo que se ejecuta para emitir informacion referente al inicio
        * de la aplicacion
        */
      iniciar(){
        this.$emit('iniciar', true)
      }
    },
}