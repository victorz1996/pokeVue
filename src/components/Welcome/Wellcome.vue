<template>
  <div class="welcome">
    <div class="row">
      <div class="col-sm-12 d-flex justify-content-center aling-items-center">
        <img class="img" src="pikachu.png" />
      </div>
      <div class="col-sm-12 d-flex justify-content-center aling-items-center">
        <h5 class="mt-5">Welcome to Pokédex</h5>
      </div>
      <div class="col-sm-12 d-flex justify-content-center aling-items-center">
        <p class="text-center mt-3">
          The digital encyclopedia created by Professor Oak is an invaluable
          tool <br />
          to Trainers in the Pokémon world.
        </p>
      </div>
    </div>
    <div
      class="col-sm-12 mt-4 d-flex justify-content-center aling-items-center"
    >
      <button @click="iniciar" type="button" class="btn boton">Get started</button>
    </div>
  </div>
</template>
<script src="./Wellcome.js"></script>
<style lang="css" src="./Wellcome.css"></style>