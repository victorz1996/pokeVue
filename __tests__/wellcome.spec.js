import { shallowMount } from '@vue/test-utils'
import Wellcome from "../src/components/Welcome/Wellcome.vue"

let cmp

    // eslint-disable-next-line no-undef
    beforeEach(() => {
      cmp = shallowMount(Wellcome)
    })
    // eslint-disable-next-line no-undef
    describe('Componente Wellcome', () => {
      // eslint-disable-next-line no-undef
      test('Instancia de Vue.js',() => {
        // eslint-disable-next-line no-undef
        expect(cmp.isVueInstance).toBeTruthy()
      })

      // eslint-disable-next-line no-undef
      it('HTMl renderizado structure esperada',()=>{
        // eslint-disable-next-line no-undef
        expect(cmp.element).toMatchSnapshot()
    })
})