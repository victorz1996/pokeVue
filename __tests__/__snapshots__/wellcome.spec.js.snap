// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Wellcome HTMl renderizado structure esperada 1`] = `
<div
  class="welcome"
>
  <div
    class="row"
  >
    <div
      class="col-sm-12 d-flex justify-content-center aling-items-center"
    >
      <img
        class="img"
        src="pikachu.png"
      />
    </div>
     
    <div
      class="col-sm-12 d-flex justify-content-center aling-items-center"
    >
      <h5
        class="mt-5"
      >
        Welcome to Pokédex
      </h5>
    </div>
     
    <div
      class="col-sm-12 d-flex justify-content-center aling-items-center"
    >
      <p
        class="text-center mt-3"
      >
        
        The digital encyclopedia created by Professor Oak is an invaluable
        tool 
        <br />
        
        to Trainers in the Pokémon world.
      
      </p>
    </div>
  </div>
   
  <div
    class="col-sm-12 mt-4 d-flex justify-content-center aling-items-center"
  >
    <button
      class="btn boton"
      type="button"
    >
      Get started
    </button>
  </div>
</div>
`;
